client/src/components/User/getUserData/index.js

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from "react";
```

Comment:
Since this is a custom hook instead of a component, it would make sense to move this to a hooks folder instead of the components folder

Reply:
It would also be good to align the names of the file and the default export. I recommend having the name start with `use` as it's a custom hook